pints/_boundaries.py

```diff
@@ -13,6 +13,38 @@
 
 
 class Boundaries(object):
+    """
+    Abstract class representing boundaries on a parameter space.
+    """
+    def check(self, parameters):
+        """
+        Returns ``True`` if and only if the given point in parameter space is
+        within the boundaries.
+        """
+        raise NotImplementedError
+
+    def n_parameters(self):
+        """
+        Returns the dimension of the parameter space these boundaries are
+        defined on.
+        """
+        raise NotImplementedError
+
+    def sample(self, n=1):
+        """
+        Returns ``n`` random samples from within the boundaries.
+
+        The returned value is a numpy array with shape ``(n, d)`` where ``n``
+        is the requested number of samples, and ``d`` is the dimension of the
+        parameter space these boundaries are defined on.
+
+        *Note that the :meth:`sample()` method does not always have to be
```

Comment:
might want to specify the cases in which the `sample` method *does have to be implemented